launcher/src/backend/ethereum-services/ExternalConsensusService.js

```diff
@@ -0,0 +1,52 @@
+import { NodeService } from "./NodeService";
+import { ServiceVolume } from "./ServiceVolume";
+
+export class ExternalConsensusService extends NodeService {
+  static buildByUserInput(network, dir, source) {
+    const service = new ExternalConsensusService();
+    service.setId();
+
+    const workingDir = service.buildWorkingDir(dir);
+    const volumes = [new ServiceVolume(workingDir + "/link.txt", "")];
+    service.init(
+      "ExternalConsensusService", // service
+      service.id, // id
+      1, // configVersion
+      [], // image
+      null, // imageVersion
+      [], // command
+      [], // entrypoint
+      { link: source }, // env
+      [], // ports
+      volumes, // volumes
+      null, // user
+      network, // network
+      null, // executionClients
+      null, // consensusClients
+      null // mevboost
+    );
+    return service;
+  }
+
+  buildConsensusClientHttpEndpointUrl(val) {
```

Comment:
this is not necessary.
you should check for `this.env.link` and return that

Reply:
simply return `""` if `this.env.link` is not set

Reply:
yea agree, first solution was with some global variable, thats why I still trying to put some value into the function:). improved! Thanks!